src/main/java/com/thealgorithms/others/CountWords.java

```diff
@@ -1,46 +1,18 @@
-package com.thealgorithms.others;
+public class Solution {
 
-/**
- * @author Marcus
- */
-final public class CountWords {
-    private CountWords() {
-    }
+	public static int countWords(String str) {	
+		int start=0;
+		int end=str.length();
+		int count=1;
+		while(start<end){
+			if(str.charAt(start)==' ' && str.charAt(start+1)!=' '){ //simple solution looking for space if we found it is considered a word
```

Comment:
As per existing code , it was checking for all non alphanumeric characters for word splitting, I think you should include that check in your condition instead of just space.